### PR TITLE
Add implementation of XPath function fn:tokenize (for formula linkbase to use)

### PR DESCRIPTION
**Pull request by strr**
#### Reason for change
Arelle is missing support for some XPath 2.0 standard functions and there is a good chance that an upcoming taxonomy might use [`fn:tokenize()`](https://www.w3.org/TR/xpath-functions/#func-tokenize) so add an implementation of it. 

#### Description of change
Implement `fn:tokenize()` and add a few tests for it.

#### Steps to Test
A simple formula linkbase making use of tokenize (zipped purely to get past Github upload filter,)
[test-tokenize-formula.zip](https://github.com/user-attachments/files/16175679/test-tokenize-formula.zip)

```
python arelleCmdLine.py -v -f test-tokenize-formula.xml
```

Without this pull request it fails:
```
[info] loaded in 0.03 secs at 2024-07-11T11:48:35 - C:\b\test-tokenize-formula.xml
[] Formula xpath2 grammar initialized in 0.63 secs -
[err:XPST0017] Variable set AssertionTokenizeAndJoin
Exception: Function named fn:tokenize does not have a custom or built-in implementation. - test-tokenize-formula.xml 14
[err:XPST0017] Variable set AssertionCountTokensZeroMatch
Exception: Function named fn:tokenize does not have a custom or built-in implementation. - test-tokenize-formula.xml 11
[err:XPST0017] Variable set AssertionCountTokens
Exception: Function named fn:tokenize does not have a custom or built-in implementation. - test-tokenize-formula.xml 8
[info] validated in 0.66 secs - C:\b\test-tokenize-formula.xml
```

With this pull request it succeeds:
```
[info] loaded in 0.03 secs at 2024-07-11T11:53:45 - C:\b\test-tokenize-formula.xml
[] Formula xpath2 grammar initialized in 1.01 secs -
[message:AssertionTokenizeAndJoin] Turn '  a b  c   ' into 'abc' 'abc'  - test-tokenize-formula.xml 14
[message:AssertionCountTokens] Length of count(fn:tokenize('a-b-c-d-e-f-g', '-')) should be 7 and actually is '7'  - test-tokenize-formula.xml 8
[message:AssertionCountTokensZeroMatch] Length of count(fn:tokenize('a-b-c-d-e-f-g-', '-')) should be 8 and actually is '8'  - test-tokenize-formula.xml 11
[info] validated in 1.05 secs - C:\b\test-tokenize-formula.xml
```

Alternately, the unit tests can be run:
```
pytest .\tests\unit_tests\arelle\test_functionfn.py
```

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by richieeeeeeee** [comment]
Thank you Stuart, this will be essential for the ESRS XBRL taxonomy.

File: arelle/FunctionFn.py [hunk 1]
```diff
@@ -307,8 +307,34 @@ def replace(xc, p, contextItem, args):
...
+
+    flags = regexFlags(xc, p, args, 2)
+    flags |= re.V1 # Enable zero width matches
+
+    m_flags = "" if len(args) < 3 else args[2]
+    if matches(xc, p, contextItem, ["", pattern, m_flags]):
+        raise XPathContext.XPathException(p, 'err:FORX0003', _('fn:tokenize $pattern matches a zero-length string: {0}').format(pattern))
+
+    try:
+        return re.split(pattern, input, flags=flags)
```
Comment (austinmatherne-wk) [inline arelle/FunctionFn.py:334]: Hi Stuart,

Thank you for your contribution! It was great talking to you at XBRL Europe. I hope your journey home was smooth.

I believe we can handle capturing groups with the following code, but I'm also fine with accepting your current submission and addressing this in a follow-up:
```suggestion
        result = []
        lastEnd = 0
        for match in re.finditer(pattern, input, flags=flags):
            start, end = match.span()
            result.append(input[lastEnd:start])
            lastEnd = end
        result.append(input[lastEnd:])
        return result
```

**Review by austinmatherne-wk — APPROVED** [review]

**Review by aaroncameron-wk — DISMISSED** [review]

File: arelle/FunctionFn.py (hunk 1 above)
Comment (strr) [reply arelle/FunctionFn.py]: Hi Austin,

It was a great conversation! Yeah all well thanks and I hope your own travels go well this summer.

That's a great suggestion. I've just pushed a further commit incorporating it.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**